src/main/java/com/cqse/teamscaleupload/ReportPatternManager.java

```diff
@@ -13,23 +14,49 @@
 import java.util.stream.Collectors;
 
 /**
- * This class manages the given file formats and
- * corresponding report file patterns.
+ * This class provides the functionality to resolve the report file patterns and formats
+ * from the command line to a map from report formats to actual files which can then be
+ * used for the upload to Teamscale.
  */
 public class ReportPatternManager {
 
-
-    /** A map from file formats to corresponding report file patterns. */
-    private final Map<String, Set<String>> formatToFileNames = new HashMap<>();
-
     /** Expected pattern for formats in the input file, matches e.g. "[vs_coverage]" */
     private static final Pattern FORMAT_PATTERN = Pattern.compile("\\[(\\w+)\\]");
 
     /**
-     * Reads all file patterns which are specified in the given input file
-     * and adds them to {@link #formatToFileNames} for the respective format.
+     * Returns a map from file formats to corresponding report files.
+     *
+     * <ol>
+     *     <li>Reads file patterns and formats from the input file</li>
+     *     <li>Reads file patterns for the format specified on the command line</li>
+     *     <li>Resolves all file patterns from the previous steps to actual files.</li>
+     * </ol>
      */
-    public void addReportFilePatternsFromInputFile(Path input) throws IOException {
+    public static Map<String, Set<File>> resolveInputFilePatters(Path inputFile, List<String> filePatterns, String format)
+            throws IOException, AgentOptionParseException {
+        Map<String, Set<String>> formatToFilePatterns = new HashMap<>();
+
+        if (inputFile != null) {
+            formatToFilePatterns = addReportFilePatternsFromInputFile(inputFile);
+        }
+        addFilePatternsForFormatOnCommandLine(formatToFilePatterns, filePatterns, format);
+
+        // Resolve all file patterns to the actual files for all given formats.
+        Map<String, Set<File>> formatToFiles = new HashMap<>();
+        for (String key : formatToFilePatterns.keySet()) {
+            Set<String> patternsForFormat = formatToFilePatterns.get(key);
+            formatToFiles.put(format, resolveFilesForPatterns(patternsForFormat));
+        }
+
+        return formatToFiles;
+    }
+
+    /**
+     * Returns a map from report format to file patterns which are read from the given input file.
+     * In the end, it is verified that no format has an empty set of patterns ({@link #validatePatternsForFormats(Map)})
+     */
+    private static Map<String, Set<String>> addReportFilePatternsFromInputFile(Path input) throws IOException {
```

Comment:
`add` seems weird since this doesn't add anything anymore. Instead it returns a map. Can you rename the function please? E.g. `parseInputFile` or something along those lines.